packages/bot/src/scratch/hooks/block.js

```diff
@@ -112,3 +112,7 @@ Blockly.Block.getDimensions = function(block_node) {
 
     return block_hw;
 };
+
+Blockly.Block.isDynamic = function(block_type) {
+    return /^((procedures_)|(variables_)|(math_change$))/.test(block_type);
```

Comment:
The `$` character in your regex pattern should be outside the `math_change` group.

Reply:
`$` only apply to math_change, because we only need `procedures_` & `variables_` as prefix